jgrapht-core/src/main/java/org/jgrapht/util/VertexToIntegerMapping.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.util;
+
+import java.util.*;
+
+/**
+ * Helper class for building a one-to-one mapping for a collection of vertices to the integer range $[0, n)$
+ * where $n$ is the number of unique vertices in the collection.
```

Comment:
We should probably add that this class only produces the desired lists/maps once, i.e. it doesn't do live updates.